### PR TITLE
[Merged by Bors] - Bump stacks and demos to 23.4 and parameterize them

### PR DESCRIPTION
**Pull request by sbernauer**
## Description

To test run (from feature branch)
`stackablectl --additional-stacks-file stacks/stacks-v2.yaml --additional-demos-file demos/demos-v2.yaml demo list`

Criteria
- [x] Bump all product versions
- [x] Bump all helm-charts, such as postgres, redis, minio etc.
- [x] Parametrize stacks (and demos), so that at least public credentials (e.g. admin) can be changed to secure values

## Review Checklist
- [x] Code contains useful comments
- [x] (Integration-)Test cases added (or not applicable)
- [x] Documentation added (or not applicable)
- [x] Changelog updated (or not applicable)
- [x] Cargo.toml only contains references to git tags (not specific commits or branches)

Once the review is done, comment `bors r+` (or `bors merge`) to merge. [Further information](https://bors.tech/documentation/getting-started/#reviewing-pull-requests)

### CODE REVIEW

File: demos/trino-taxi-data/setup-superset.yaml [hunk 1]
```diff
@@ -9,14 +9,24 @@ spec:
       containers:
         - name: setup-superset
           image: docker.stackable.tech/stackable/testing-tools:0.1.0-stackable0.1.0
-          command: ["bash", "-c", "curl -o superset-assets.zip https://raw.githubusercontent.com/stackabletech/stackablectl/main/demos/trino-taxi-data/superset-assets.zip && python -u /tmp/script/script.py"]
+          command: ["bash", "-c", "curl -o superset-assets.zip https://raw.githubusercontent.com/stackabletech/stackablectl/release-23.4/demos/trino-taxi-data/superset-assets.zip && python -u /tmp/script/script.py"]
```
Comment (sbernauer) [inline demos/trino-taxi-data/setup-superset.yaml:12]: Marker

File: demos/spark-k8s-anomaly-detection-taxi-data/setup-superset.yaml [hunk 2]
```diff
@@ -9,14 +9,24 @@ spec:
       containers:
         - name: setup-superset
           image: docker.stackable.tech/stackable/testing-tools:0.1.0-stackable0.1.0
-          command: ["bash", "-c", "curl -o superset-assets.zip https://raw.githubusercontent.com/stackabletech/stackablectl/main/demos/spark-k8s-anomaly-detection-taxi-data/superset-assets.zip && python -u /tmp/script/script.py"]
+          command: ["bash", "-c", "curl -o superset-assets.zip https://raw.githubusercontent.com/stackabletech/stackablectl/release-23.4/demos/spark-k8s-anomaly-detection-taxi-data/superset-assets.zip && python -u /tmp/script/script.py"]
```
Comment (sbernauer) [inline demos/spark-k8s-anomaly-detection-taxi-data/setup-superset.yaml:12]: Marker

File: demos/data-lakehouse-iceberg-trino-spark/setup-superset.yaml [hunk 3]
```diff
@@ -9,14 +9,24 @@ spec:
       containers:
         - name: setup-superset
           image: docker.stackable.tech/stackable/testing-tools:0.1.0-stackable0.1.0
-          command: ["bash", "-c", "curl -o superset-assets.zip https://raw.githubusercontent.com/stackabletech/stackablectl/main/demos/data-lakehouse-iceberg-trino-spark/superset-assets.zip && python -u /tmp/script/script.py"]
+          command: ["bash", "-c", "curl -o superset-assets.zip https://raw.githubusercontent.com/stackabletech/stackablectl/release-23.4/demos/data-lakehouse-iceberg-trino-spark/superset-assets.zip && python -u /tmp/script/script.py"]
```
Comment (sbernauer) [inline demos/data-lakehouse-iceberg-trino-spark/setup-superset.yaml:12]: Marker

File: demos/data-lakehouse-iceberg-trino-spark/create-nifi-ingestion-job.yaml [hunk 4]
```diff
@@ -8,25 +8,30 @@ spec:
...
       initContainers:
-        - name: wait-for-testdata 
+        - name: wait-for-kafka
           image: docker.stackable.tech/stackable/tools:0.2.0-stackable0.3.0
           command: ["bash", "-c", "echo 'Waiting for all kafka brokers to be ready' && kubectl wait --for=condition=ready --timeout=30m pod -l app.kubernetes.io/instance=kafka -l app.kubernetes.io/name=kafka"]
       containers:
         - name: create-nifi-ingestion-job
           image: docker.stackable.tech/stackable/testing-tools:0.1.0-stackable0.1.0
-          command: ["bash", "-c", "curl -O https://raw.githubusercontent.com/stackabletech/stackablectl/main/demos/data-lakehouse-iceberg-trino-spark/LakehouseKafkaIngest.xml && python -u /tmp/script/script.py"]
+          command: ["bash", "-c", "curl -O https://raw.githubusercontent.com/stackabletech/stackablectl/release-23.4/demos/data-lakehouse-iceberg-trino-spark/LakehouseKafkaIngest.xml && python -u /tmp/script/script.py"]
```
Comment (sbernauer) [inline demos/data-lakehouse-iceberg-trino-spark/create-nifi-ingestion-job.yaml:17]: Marker

**Review by adwk67 — APPROVED** [review]
LGTM - just minor comments. Great job on improvements and general cleanup! I successfully ran a subset of demos locally:

- airflow
- spark anomaly detection
- hdfs/hbase cycling data
- jupyterhub anomaly detection

It might be worthwhile to bump `docker.stackable.tech/stackable/tools:0.2.0-stackable0.3.0` to `docker.stackable.tech/stackable/tools:1.0.0-stackable23.4` e.g.here: https://github.com/stackabletech/stackablectl/pull/251/files#diff-273bb9c279d3e3e4f36344ea49115f44eb320f74831ae72792adc7e5b089ef89R11
and here: https://github.com/stackabletech/stackablectl/pull/251/files#diff-1e07180c214382d6e2d1261bfcc887d36149816490441319cfb604a091e96fffL11

(or maybe just update in main so as not to have test anything again).

File: demos/jupyterhub-pyspark-hdfs-anomaly-detection-taxi-data/Dockerfile [hunk 5]
```diff
@@ -1,6 +1,6 @@
 # docker build -t docker.stackable.tech/demos/jupyter:pyspark-3.3.0-java17-stackable23.4.0-rc1 -f Dockerfile .
 
-FROM docker.stackable.tech/stackable/pyspark-k8s:3.3.0-java17-stackable23.4.0-rc1
+FROM docker.stackable.tech/stackable/pyspark-k8s:3.3.0-stackable23.4
```
[the remark this answers is not in the document]
Comment (sbernauer) [reply demos/jupyterhub-pyspark-hdfs-anomaly-detection-taxi-data/Dockerfile:3]: Adressed offline

**Pull request closed without merging**